src/index.js

```diff
@@ -1,207 +1,210 @@
-    config: PropTypes.object,
-    gdprAppliesGlobally: PropTypes.bool,
-    onReady: PropTypes.func,
-    onConsentChanged: PropTypes.func,
-    onNoticeShown: PropTypes.func,
-    onNoticeHidden: PropTypes.func,
-    onNoticeBackdropclick: PropTypes.func,
-    onNoticeClickAgree: PropTypes.func,
-    onNoticeClickMoreInfo: PropTypes.func,
-    onPreferencesClickAgreeToAll: PropTypes.func,
-    onPreferencesClickDisagreeToAll: PropTypes.func,
-    onPreferencesClickPurposeAgree: PropTypes.func,
-    onPreferencesClickPurposeDisagree: PropTypes.func,
-    onPreferencesClickViewVendors: PropTypes.func,
-    onPreferencesClickSaveChoices: PropTypes.func,
-    onPreferencesClickVendorAgree: PropTypes.func,
-    onPreferencesClickVendorDisagree: PropTypes.func,
-    onPreferencesClickVendorSaveChoices: PropTypes.func
-  }
-
-  static defaultProps = {
-    apiKey: null,
-    iabVersion: 1,
-    noticeId: null,
-    config: {},
-    gdprAppliesGlobally: true,
-    onReady: () => {},
-    onConsentChanged: () => {},
-    onNoticeShown: () => {},
-    onNoticeHidden: () => {},
-    onNoticeBackdropclick: () => {},
-    onNoticeClickAgree: () => {},
-    onNoticeClickMoreInfo: () => {},
-    onPreferencesClickAgreeToAll: () => {},
-    onPreferencesClickDisagreeToAll: () => {},
-    onPreferencesClickPurposeAgree: () => {},
-    onPreferencesClickPurposeDisagree: () => {},
-    onPreferencesClickViewVendors: () => {},
-    onPreferencesClickSaveChoices: () => {},
-    onPreferencesClickVendorAgree: () => {},
-    onPreferencesClickVendorDisagree: () => {},
-    onPreferencesClickVendorSaveChoices: () => {}
-  }
-
+const DidomiSDK = ({
+  apiKey: apiKeyProp =  null,
+  iabVersion =  2,
+  noticeId =  null,
+  config =  {},
+  gdprAppliesGlobally: gdprAppliesGloballyProp =  true,
+  onReady =  () => {},
+  onConsentChanged =  () => {},
+  onNoticeShown =  () => {},
+  onNoticeHidden =  () => {},
+  onNoticeBackdropclick =  () => {},
+  onNoticeClickAgree =  () => {},
+  onNoticeClickMoreInfo =  () => {},
+  onPreferencesClickAgreeToAll =  () => {},
+  onPreferencesClickDisagreeToAll =  () => {},
+  onPreferencesClickPurposeAgree =  () => {},
+  onPreferencesClickPurposeDisagree =  () => {},
+  onPreferencesClickViewVendors =  () => {},
+  onPreferencesClickSaveChoices =  () => {},
+  onPreferencesClickVendorAgree =  () => {},
+  onPreferencesClickVendorDisagree =  () => {},
+  onPreferencesClickVendorSaveChoices =  () => {},
+  sdkPath = "https://sdk.privacy-center.org/"
+}) => {
+ 
   /**
    * Called once the Didomi SDK is ready and loaded
    * @param {*} Didomi
    */
-  didomiOnReady(Didomi) {
-    this.props.onReady(Didomi)
-    this.setEvents(Didomi);
+  const handleDidomiOnReady = (Didomi) => {
+    onReady(Didomi)
+    setEvents(Didomi);
   }
 
   /**
    * Set all the Didomi events the return the callbacks from the props
    * @param {*} Didomi
    */
-  setEvents(Didomi) {
-    if(this.props.onConsentChanged) {
+  const setEvents = (Didomi) => {
```

Comment:
Is there also a need to unregister events when component is unmounted ?

Reply:
Yes but I'd rather do it in a separate PR rather than this one as it's starting to be pretty big.
Also, it's kind of a theoretical case as you are not supposed to render the `DidomiSDK` more than once a page and there is no reason to unmount it.

We'll track this in https://github.com/didomi/react/issues/26